clients/filesystem-hadoop/src/main/java/com/datastrato/gravitino/filesystem/hadoop/GravitinoVirtualFileSystem.java

```diff
@@ -0,0 +1,406 @@
+    if (meta.getFileSystem().isFile(new Path(meta.getFileset().storageLocation()))) {
+      throw new UnsupportedOperationException(
+          String.format(
+              "Cannot rename the fileset: %s which only mounts a single file.", srcIdentifier));
+    }
+
+    Path srcActualPath = resolvePathByIdentifier(srcIdentifier, meta, src);
+    Path dstActualPath = resolvePathByIdentifier(dstIdentifier, meta, dst);
+    return meta.getFileSystem().rename(srcActualPath, dstActualPath);
+  }
+
+  @Override
+  public boolean delete(Path f, boolean recursive) throws IOException {
+    NameIdentifier identifier = extractIdentifier(f.toUri());
+    FilesetMeta meta = getCachedFileset(identifier);
+    Path actualPath = resolvePathByIdentifier(identifier, meta, f);
+    return meta.getFileSystem().delete(actualPath, recursive);
+  }
+
+  @Override
+  public FileStatus[] listStatus(Path f) throws IOException {
+    NameIdentifier identifier = extractIdentifier(f.toUri());
+    FilesetMeta meta = getCachedFileset(identifier);
+    Path actualPath = resolvePathByIdentifier(identifier, meta, f);
+    FileStatus[] fileStatusResults = meta.getFileSystem().listStatus(actualPath);
+    return Arrays.stream(fileStatusResults)
+        .map(
+            fileStatus ->
+                resolveFileStatusPathPrefix(
+                    fileStatus,
+                    meta.getFileset().storageLocation(),
+                    concatFilesetPrefix(identifier, meta)))
+        .toArray(FileStatus[]::new);
+  }
+
+  @Override
+  public void setWorkingDirectory(Path newDir) {
+    try {
+      NameIdentifier identifier = extractIdentifier(newDir.toUri());
+      FilesetMeta meta = getCachedFileset(identifier);
+      Path actualPath = resolvePathByIdentifier(identifier, meta, newDir);
+      meta.getFileSystem().setWorkingDirectory(actualPath);
+      this.workingDirectory = newDir;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public Path getWorkingDirectory() {
+    return this.workingDirectory;
+  }
+
+  @Override
+  public boolean mkdirs(Path f, FsPermission permission) throws IOException {
+    NameIdentifier identifier = extractIdentifier(f.toUri());
+    FilesetMeta meta = getCachedFileset(identifier);
+    Path actualPath = resolvePathByIdentifier(identifier, meta, f);
+    return meta.getFileSystem().mkdirs(actualPath, permission);
+  }
+
+  @Override
+  public FileStatus getFileStatus(Path f) throws IOException {
+    NameIdentifier identifier = extractIdentifier(f.toUri());
+    FilesetMeta meta = getCachedFileset(identifier);
+    Path actualPath = resolvePathByIdentifier(identifier, meta, f);
+    FileStatus fileStatus = meta.getFileSystem().getFileStatus(actualPath);
+    return resolveFileStatusPathPrefix(
+        fileStatus, meta.getFileset().storageLocation(), concatFilesetPrefix(identifier, meta));
+  }
+
+  private String concatFilesetPrefix(NameIdentifier identifier, FilesetMeta meta) {
```

Comment:
This method seems only get the fileset prefix, doesn't do any concatenation thing, so the method name should be changed.